### PR TITLE
add memory optimization design doc

### LINKED ISSUE #7205: Design Doc on memory optimization

### PR DESCRIPTION
**Pull request by QiJune**
Fix #7205

### CODE REVIEW

File: doc/design/memory_optimization.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+- availability of Big Data
+- supercomputing power to process this Big Data over very large neural networks
+- modern algorithms
+
+Following graph shows the details:
+
+![](images/deep_learning.png)
+
+Larger model usually brings better performance. However, GPU memory is certain limited. For example, the memory size of a GTX TITAN X is only 12GB. To train complex and large model, we have to take care of memory using. Besides, memory optimization is also necessary in both online/mobile inference.
```
Comment (abhinavarora) [inline doc/design/memory_optimization.md]: model -> **models**

Comment (abhinavarora) [reply doc/design/memory_optimization.md]: It should be `Larger **models** usually **bring**`

Comment (abhinavarora) [reply doc/design/memory_optimization.md]: It should be `However, GPU memory is limited`. We can drop the **certain**

Comment (QiJune) [reply doc/design/memory_optimization.md]: Thanks to @abhinavarora , I will merge it first.
@luotao1 The graphs will polished in next PR.

**Pull request merged**